### DISCUSSION TITLE
Sonoff S31 - Continual Software Restarts

### DISCUSSION THREAD

**Discussion by coolspot18** [arendst/Tasmota#17058] [Q&A] [answered]
Hi All,

I just flashed two Sonoff S31s with Tasmota, but the units are going into a constant "Software Restart" cycle every 5 to 10 minutes. 

Any ideas why this is happening?

The logs don't show much - running 12.2.0, should I try the development branch?

I do have MQTT enabled to use with Home Assistant and I'm seeing a lot of errors in the MQTT logs. WiFi signal strength is about 85%+, but I do have trouble accessing the web interface. But poor WiFi shouldn't cause the switch to crash continually?

`2022-11-13 12:18:24: Warning: Received PUBREC from DVES_282BB9 for an unknown packet identifier 26996.
2022-11-13 12:18:24: Client DVES_282BB9 disconnected due to protocol error.
2022-11-13 12:18:37: New connection from 192.168.0.150:61930 on port 1883.
2022-11-13 12:18:37: New client connected from 192.168.0.150:61930 as DVES_282BB9 (p2, c1, k30, u'tasmota').
2022-11-13 12:19:07: Bad client DVES_282BB9 sending multiple CONNECT messages.
2022-11-13 12:19:07: Client DVES_282BB9 disconnected due to protocol error.`

I can't seem to find much as to why the switch would continually crash, I thought the S31 was well supported?

Thanks!

`18:23:29.761 MQT: stat/tasmota_282BB9/STATUS12 = {"StatusSTK":{"Exception":4,"Reason":"Software Watchdog","EPC":["40003b3c","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000"}}
18:23:31.865 MQT: stat/tasmota_282BB9/STATUS = {"Status":{"Module":41,"DeviceName":"tasmota_282BB9","FriendlyName":["Rec Room TV Wall Outlet 2"],"Topic":"tasmota_282BB9","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
18:23:31.871 MQT: stat/tasmota_282BB9/STATUS1 = {"StatusPRM":{"Baudrate":4800,"SerialConfig":"8E1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software Watchdog","Uptime":"0T00:00:33","StartupUTC":"2022-11-13T17:22:58","Sleep":50,"CfgHolder":4617,"BootCount":11,"BCResetTime":"2022-11-13T09:25:03","SaveCount":31,"SaveAddress":"F6000"}}
18:23:31.875 MQT: stat/tasmota_282BB9/STATUS2 = {"StatusFWR":{"Version":"12.2.0(tasmota)","BuildDateTime":"2022-10-17T08:39:23","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"372/699"}}
18:23:31.880 MQT: stat/tasmota_282BB9/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["WifINet",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000680003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
18:23:31.894 MQT: stat/tasmota_282BB9/STATUS4 = {"StatusMEM":{"ProgramSize":626,"Free":376,"Heap":20,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
18:23:31.901 MQT: stat/tasmota_282BB9/STATUS5 = {"StatusNET":{"Hostname":"tasmota-282BB9-3001","IPAddress":"192.168.0.150","Gateway":"192.168.1.1","Subnetmask":"255.255.254.0","DNSServer1":"192.168.1.100","DNSServer2":"192.168.1.101","Mac":"48:3F:DA:28:2B:B9","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
18:23:31.905 MQT: stat/tasmota_282BB9/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.40","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_282BB9","MqttUser":"tasmota","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
18:23:31.911 MQT: stat/tasmota_282BB9/STATUS7 = {"StatusTIM":{"UTC":"2022-11-13T17:23:31","Local":"2022-11-13T18:23:31","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"07:56","Sunset":"17:12"}}
18:23:31.914 MQT: stat/tasmota_282BB9/STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
18:23:31.924 MQT: stat/tasmota_282BB9/STATUS10 = {"StatusSNS":{"Time":"2022-11-13T18:23:31","ENERGY":{"TotalStartTime":"2022-11-13T09:25:03","Total":0.156,"Yesterday":0.000,"Today":0.156,"Power":4,"ApparentPower":19,"ReactivePower":18,"Factor":0.20,"Voltage":122,"Current":0.151}}}
18:23:31.985 MQT: stat/tasmota_282BB9/STATUS11 = {"StatusSTS":{"Time":"2022-11-13T18:23:31","Uptime":"0T00:00:33","UptimeSec":33,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"WifINet","BSSId":"1E:61:B4:AE:B1:A6","Channel":3,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:06"}}}
18:23:31.990 MQT: stat/tasmota_282BB9/STATUS12 = {"StatusSTK":{"Exception":4,"Reason":"Software Watchdog","EPC":["40003b3c","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000"}}
18:23:39.905 MQT: stat/tasmota_282BB9/STATUS = {"Status":{"Module":41,"DeviceName":"tasmota_282BB9","FriendlyName":["Rec Room TV Wall Outlet 2"],"Topic":"tasmota_282BB9","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
18:23:39.911 MQT: stat/tasmota_282BB9/STATUS1 = {"StatusPRM":{"Baudrate":4800,"SerialConfig":"8E1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software Watchdog","Uptime":"0T00:00:41","StartupUTC":"2022-11-13T17:22:58","Sleep":50,"CfgHolder":4617,"BootCount":11,"BCResetTime":"2022-11-13T09:25:03","SaveCount":31,"SaveAddress":"F6000"}}
18:23:39.915 MQT: stat/tasmota_282BB9/STATUS2 = {"StatusFWR":{"Version":"12.2.0(tasmota)","BuildDateTime":"2022-10-17T08:39:23","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"372/699"}}
18:23:39.919 MQT: stat/tasmota_282BB9/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["WifINet",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000680003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
18:23:39.931 MQT: stat/tasmota_282BB9/STATUS4 = {"StatusMEM":{"ProgramSize":626,"Free":376,"Heap":20,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
18:23:39.937 MQT: stat/tasmota_282BB9/STATUS5 = {"StatusNET":{"Hostname":"tasmota-282BB9-3001","IPAddress":"192.168.0.150","Gateway":"192.168.1.1","Subnetmask":"255.255.254.0","DNSServer1":"192.168.1.100","DNSServer2":"192.168.1.101","Mac":"48:3F:DA:28:2B:B9","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
18:23:39.941 MQT: stat/tasmota_282BB9/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.40","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_282BB9","MqttUser":"tasmota","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
18:23:39.949 MQT: stat/tasmota_282BB9/STATUS7 = {"StatusTIM":{"UTC":"2022-11-13T17:23:39","Local":"2022-11-13T18:23:39","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"07:56","Sunset":"17:12"}}
18:23:39.954 MQT: stat/tasmota_282BB9/STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
18:23:39.964 MQT: stat/tasmota_282BB9/STATUS10 = {"StatusSNS":{"Time":"2022-11-13T18:23:39","ENERGY":{"TotalStartTime":"2022-11-13T09:25:03","Total":0.156,"Yesterday":0.000,"Today":0.156,"Power":4,"ApparentPower":18,"ReactivePower":18,"Factor":0.20,"Voltage":122,"Current":0.151}}}
18:23:39.990 MQT: stat/tasmota_282BB9/STATUS11 = {"StatusSTS":{"Time":"2022-11-13T18:23:39","Uptime":"0T00:00:41","UptimeSec":41,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":25,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"WifINet","BSSId":"1E:61:B4:AE:B1:A6","Channel":3,"Mode":"11n","RSSI":86,"Signal":-57,"LinkCount":1,"Downtime":"0T00:00:06"}}}
18:23:39.994 MQT: stat/tasmota_282BB9/STATUS12 = {"StatusSTK":{"Exception":4,"Reason":"Software Watchdog","EPC":["40003b3c","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000"}}
`

**Comment by sfromis** [comment, marked as answer]
Stop running TasmoAdmin. That tool is a known troublemaker to borderline devices.

**Reply by coolspot18** [reply]
> Stop running TasmoAdmin. That tool is a known troublemaker to borderline devices.

Ah ... let me try that. I was a bit suspicious about TasmoAdmin.

When you say " borderline" devices, do you mean WiFi reception?

UPDATE: Well, since turning off TasmoAdmin, the switches haven't crashed, so I think that might be the culprit...

**Reply by sfromis** [reply]
Borderline was in respect to the internal power supply being (or not) up to the task of powering extra wifi activity, which is a known malady for at least (but not limited to) Sonoff Basic.
